packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -111,18 +144,35 @@ abstract class FlutterCommand extends Command<Null> {
   /// and [runCommand] to execute the command
   /// so that this method can record and report the overall time to analytics.
   @override
-  Future<Null> run() {
-    final Stopwatch stopwatch = new Stopwatch()..start();
-    final UsageTimer analyticsTimer = usagePath == null ? null : flutterUsage.startTimer(name);
+  Future<Null> run() async {
+    final DateTime startTime = clock.now();
 
     if (flutterUsage.isFirstRun)
       flutterUsage.printWelcome();
 
-    return verifyThenRunCommand().whenComplete(() {
-      final int ms = stopwatch.elapsedMilliseconds;
-      printTrace("'flutter $name' took ${ms}ms.");
-      analyticsTimer?.finish();
-    });
+    final FlutterCommandResult commandResult = await verifyThenRunCommand();
+
+    final DateTime endTime = clock.now();
+    printTrace("'flutter $name' took ${endTime.difference(startTime).inMilliseconds}ms.");
+    if (usagePath != null) {
+      final String label = (
```

Comment:
The following phrasing is a little bit more plain, and it allow you to get rid of `_getNullableExitCodeName()` altogether.

```
final List<String> labels = <String>[];
if (commandResult?.exitCode != null)
  labels.add(getEnumName(commandResult.exitCode));
if (commandResult?.analyticsParameters?.isNotEmpty ?? false)
  labels.addAll(commandResult.analyticsParameters);
final String label = labels.join('-');
```